Lib/logging/__init__.py

```diff
@@ -569,7 +584,9 @@ def format(self, record):
         if self.usesTime():
             record.asctime = self.formatTime(record, self.datefmt)
         s = self.formatMessage(record)
-        if record.exc_info:
+        if (isinstance(record.exc_info, tuple) and record.exc_info[0]):
```

Comment:
What can it be if not a tuple?

Reply:
Boolean.

Reply:
According to the documentation, exc_info is an exception tuple with the current exception information, or None if no exception information is available. Hence `isinstance(record.exc_info, tuple)`, `record.exc_info is not None` and just `record.exc_info` should be equivalent, but the latter form is the shortest. If needed a special case for the tuple `(None, None, None)` (this is not an *empty tuple*), it should be tested.

Reply:
The documentation also says you can call log() with exc_info='a True-like value'. So calling it with exc_info=True  is also acceptable practice and examples abound. When exc_info() doesn't find a stack frame it returns (None, None, None) which you are correct is, not None. Point is Boolean does get past in and the code just assumes a Tuple will ALWAYS be the data type and blindly de-references it. The formatter also blindly assumes a valid stack frame was found and that the Tuple can't be None^3.